modules/light-clients/08-wasm/types/misbehaviour_handle_test.go

```diff
@@ -469,372 +481,128 @@ package types_test
 //				}
 //			}, false,
 //		},
-//	}
-//
-//	for _, tc := range testCases {
-//		suite.Run(tc.name, func() {
-//			suite.SetupWasmTendermint()
-//			path = ibctesting.NewPath(suite.chainA, suite.chainB)
-//
-//			err := path.EndpointA.CreateClient()
-//			suite.Require().NoError(err)
-//
-//			tc.malleate()
-//
-//			clientState := path.EndpointA.GetClientState()
-//			clientStore := suite.chainA.App.GetIBCKeeper().ClientKeeper.ClientStore(suite.chainA.GetContext(), path.EndpointA.ClientID)
-//
-//			err = clientState.VerifyClientMessage(suite.chainA.GetContext(), suite.chainA.App.AppCodec(), clientStore, misbehaviour)
-//
-//			if tc.expPass {
-//				suite.Require().NoError(err)
-//			} else {
-//				suite.Require().Error(err)
-//			}
-//		})
-//	}
+// 	}
+
+// 	for _, tc := range testCases {
+// 		suite.Run(tc.name, func() {
+// 			// reset suite to create fresh application state
+// 			suite.SetupWasmWithMockVM()
+
+// 			endpoint := wasmtesting.NewWasmEndpoint(suite.chainA)
+// 			err := endpoint.CreateClient()
+// 			suite.Require().NoError(err)
+
+// 			tc.malleate()
+
+// 			clientStore := suite.chainA.App.GetIBCKeeper().ClientKeeper.ClientStore(suite.chainA.GetContext(), endpoint.ClientID)
+// 			clientState := endpoint.GetClientState()
+// 			foundMisbehaviour := clientState.CheckForMisbehaviour(
+// 				suite.chainA.GetContext(),
+// 				suite.chainA.App.AppCodec(),
+// 				clientStore, // pass in clientID prefixed clientStore
+// 				clientMessage,
+// 			)
+
+// 			suite.Require().Equal(tc.foundMisbehaviour, foundMisbehaviour)
+// 		})
+// 	}
 // }
 
-/* func (suite *TypesTestSuite) TestCheckForMisbehaviourGrandpa() {
+func (suite *TypesTestSuite) TestCheckForMisbehaviour() {
 	var (
-		ok          bool
-		clientMsg   exported.ClientMessage
-		clientState exported.ClientState
+		foundMisbehaviour bool
+		panicErr          error
 	)
 
 	testCases := []struct {
-		name     string
-		malleate func()
-		expPass  bool
+		name                 string
+		malleate             func()
+		expFoundMisbehaviour bool
+		expPanic             bool
 	}{
 		{
-			"valid update no misbehaviour",
+			"no misbehaviour",
 			func() {
-				data, err := base64.StdEncoding.DecodeString(suite.testData["header"])
-				suite.Require().NoError(err)
-				clientMsg = &types.ClientMessage{
-					Data: data,
-				}
-
-				err = clientState.VerifyClientMessage(suite.ctx, suite.chainA.Codec, suite.store, clientMsg)
-				suite.Require().NoError(err)
+				suite.mockVM.RegisterQueryCallback(types.CheckForMisbehaviourMsg{}, func(codeID wasmvm.Checksum, env wasmvmtypes.Env, queryMsg []byte, store wasmvm.KVStore, goapi wasmvm.GoAPI, querier wasmvm.Querier, gasMeter wasmvm.GasMeter, gasLimit uint64, deserCost wasmvmtypes.UFraction) ([]byte, uint64, error) {
+					resp, err := json.Marshal(types.CheckForMisbehaviourResult{FoundMisbehaviour: false})
+					suite.Assert().NoError(err)
```

Comment:
```suggestion
					suite.Require().NoError(err)
```
Same for all Asserts.